plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergPageSourceProvider.java

```diff
@@ -454,7 +481,12 @@ private static ReaderPageSource createOrcPageSource(
         ImmutableMap.Builder<Integer, OrcColumn> columnsById = ImmutableMap.builder();
         Traverser.forTree(OrcColumn::getNestedColumns)
                 .depthFirstPreOrder(columns)
-                .forEach(column -> columnsById.put(getIcebergFieldId(column), column));
+                .forEach(column -> {
+                    String fieldId = (column.getAttributes().get(ORC_ICEBERG_ID_KEY));
+                    if (fieldId != null) {
```

Comment:
This can happen if the column does not have an id and the default name mapping is not present. @findepi suggested maybe we should fail in the case rather than having the column values be `null` https://github.com/trinodb/trino/issues/9843#issuecomment-973108743

Reply:
then keeping `getIcebergFieldId` usage seems reasonable, wdyt?

Reply:
I'm okay with returning null here, similar to Spark.